File: src/domain/value_objects/RealtimeSalesResult.js
class RealtimeSalesResult {
  constructor(asin, unitCount, totalAmount) {
    this.asin = asin;
    this.unitCount = unitCount || 0;
    this.totalAmount = totalAmount || 0;
  }

  addSale(quantity, amount) {
    this.unitCount += quantity;
    this.totalAmount += amount;
  }
}

File: py_src/domain/value_objects/realtime_sales_result.py
from dataclasses import dataclass, field


@dataclass
class RealtimeSalesResult:
    asin: str
    unit_count: int = field(default=0)
    total_amount: float = field(default=0.0)

    def add_sale(self, quantity: int, unit_price: float) -> None:
        self.unit_count += quantity
        self.total_amount += quantity * unit_price
